Add Button to Disable Auto Parse

jhoffswell opened the pull request:
Add a button to disable automatic parsing of the spec. Prevent the specification from rendering when this is set to manual. Add a button to parse the spec when in render mode.

Note: This disables the editor from rendering the visualization automatically when parse is set to 'manual' but still saves changes to the specification and shows warnings when typing.

jhoffswell committed:
Add Button to Disable Auto Parse

Add a button to disable automatic parsing of the spec. Prevent the
specification from rendering when this is set to manual. Add a button
to parse the spec when in render mode.

domoritz commented:
Looks great! How can we not update the spec/save changes to the spec until the spec is actually rendered? I think it's odd that the spec is saved before it is successfully rendered.

domoritz reviewed (changes requested):
Don't change the spec until it's being parsed.

jhoffswell committed:
Only save spec on parse

Update the editor so that it only saves the specification to the state
when the user selects ‘parse’ when in ‘manual’ mode.

jhoffswell committed:
Merge branch 'redux' into jane-autoParse

domoritz reviewed (approved):
@starry97 will merge.
